### ISSUE TITLE
Bad parsing of CMakeLists.txt

### ISSUE THREAD

**Issue by mcandre** [doxygen/doxygen#8184]
**Describe the bug**

When I try to doxygen my C project, then doxygen goes down a rabbit hole trying to parse CMakeLists.txt and fails early.

**Expected behavior**

doxygen should completely ignore the syntax in CMakeLists.txt and focus on actual source code files that contribute to a public API.

**To Reproduce**

CMakeLists.txt snippet:

```cmake
execute_process(COMMAND cat /etc/*release* OUTPUT_VARIABLE ETC_RELEASE)
```

Trace:

```console
$ doxygen Doxyfile
Searching for include files...
Searching for example files...
Searching for images...
Searching for dot files...
Searching for msc files...
Searching for dia files...
Searching for files to exclude
Searching INPUT for files to process...
Searching for files in directory /home/andrew/go/src/github.com/mcandre/toys/c/args
Reading and parsing tag files
Parsing files
Preprocessing /home/andrew/go/src/github.com/mcandre/toys/c/args/CMakeCache.txt...
Parsing file /home/andrew/go/src/github.com/mcandre/toys/c/args/CMakeCache.txt...
Preprocessing /home/andrew/go/src/github.com/mcandre/toys/c/args/CMakeLists.txt...
/home/andrew/go/src/github.com/mcandre/toys/c/args/CMakeLists.txt:117: error: Reached end of file while still inside a (nested) comment. Nesting level 1 (probable line reference: 22) (warning treated as error, aborting now)
Exiting...
```

Hint hint, `/*release*` is NOT part of a multiline comment. CMakeLists.txt uses a [completely different](https://cmake.org/cmake/help/v3.1/manual/cmake-language.7.html#comments) multiline comment syntax. In fact, doxygen shouldn't even be *opening* CMakeLists.txt, let alone trying to parse it with a C/C++ AST.

**Version**

1.8.17

As a workaround, I am skipping populating my debugging variable `ETC_RELEASE`.

**Comment by albert-github** [comment]
Doxygen does not know about the meaning of the names of files. The `*.txt` files are enabled through the `INPUT` and `FILE_PATTERNS` settings. Also of relevance is the way `.txt` files are processed i.e. as C files (this is due to historical reasons), and can be changed by means of `EXTENSION_MAPPING`.

Furthermore in doxygen files can be excluded by means of the `EXCLUDE` and `EXCLUDE_PATTERNS`. 

I think the problem is usage.

**Comment by doxygen** [comment]
@albert-github I think the problem is caused by having `*.txt` as part of the standard `FILE_PATTERNS`. This was introduced by https://github.com/doxygen/doxygen/commit/062257aee6b51deb6bfd64a5a9a3583cd333048c. I think we should remove the `*.txt` wildcard because it is not likely to be in the correct format.

**Comment by albert-github** [comment]
In the mentioned pull request the documentation has been consistent in respect to the code (`util.cpp`) and lists all the extension used here.
In the code of `util.cpp` I see:
```
  updateLanguageMapping(".txt",      "c"); // see bug 760836
```
in the issue 760836 I also see the comment:
```
On 2016-01-31 13:05:48 +0000, Dimitri van Heesch wrote:

    I'll remove parsing .txt files by default like before. Then the user can configure doxygen to do this using the "c" or "markdown" parser if needed, using FILE_PATTERNS and EXTENSION_MAPPING.

```

Problem will be that when removing the `.txt` (and `.doc`) from the list this will lead to a possible compatibility problem for users of the version 1.8.17 and upward (I was afraid for Fossies and CGAL but they properly include the `.txt` and set the `EXTENSION_MAPPING`).

**Comment by doxygen** [comment]
@albert-github if we keep the language mapping and not have `*.txt` as a default pattern in the config file it would still be compatible I think. If the `*.txt` extension is already in an existing config file without explicitly mapping it still has the backward compatible mapping, and for newly generated configuration file the `*.txt` extension will no longer appear preventing `*.txt` files from being parsed by default.

**Comment by albert-github** [comment]
@doxygen I think this will be compatible but it will also mean that the documentation for the FILE_PATTERNS and the language mapping will not 100% match  (language mapping will have an extra element). I don't know whether this is a big problem.

Should the doc extension also be removed from the FILE_PATTERNS as it pickups Microsoft word documents as well or should we here check on the first few bytes and see whether it is a Word document?

**Comment by doxygen** [comment]
@albert-github I'm not too worried about the mismatch between FILE_PATTERNS and the language mapping, and I think we should remove the doc extension as well indeed.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8200.

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
